sdk/app_cpu1/common/drv/eddy_current_sensor.c

```diff
@@ -9,36 +9,103 @@ void eddy_current_sensor_init(void)
-    eddy_current_sensor_set_sample_rate(EDDY_CURRENT_SENSOR_4_BASE_ADDR, 20000);
+    eddy_current_sensor_trigger_on_pwm_high(EDDY_CURRENT_SENSOR_2_BASE_ADDR);
+    eddy_current_sensor_set_timing(EDDY_CURRENT_SENSOR_2_BASE_ADDR, 2000, 270);
+    eddy_current_sensor_trigger_on_pwm_high(EDDY_CURRENT_SENSOR_3_BASE_ADDR);
+    eddy_current_sensor_set_timing(EDDY_CURRENT_SENSOR_3_BASE_ADDR, 2000, 270);
+    eddy_current_sensor_trigger_on_pwm_high(EDDY_CURRENT_SENSOR_4_BASE_ADDR);
+    eddy_current_sensor_set_timing(EDDY_CURRENT_SENSOR_4_BASE_ADDR, 2000, 270);
 #endif
 }
 
-void eddy_current_sensor_enable(uint32_t base_addr)
+void eddy_current_sensor_trigger_on_pwm_high(uint32_t base_addr)
 {
-    Xil_Out32(base_addr + (3 * sizeof(uint32_t)), 1);
+    // Get the current value of the config register and set the pwm_high trigger bit
+    uint32_t config_reg_address = base_addr + (3 * sizeof(uint32_t));
+    Xil_Out32(config_reg_address, (Xil_In32(config_reg_address) | 0x1));
 }
 
-void eddy_current_sensor_disable(uint32_t base_addr)
+void eddy_current_sensor_trigger_on_pwm_low(uint32_t base_addr)
 {
-    Xil_Out32(base_addr + (3 * sizeof(uint32_t)), 0);
+    // Get the current value of the config register and set the pwm_low trigger bit
+    uint32_t config_reg_address = base_addr + (3 * sizeof(uint32_t));
+    Xil_Out32(config_reg_address, (Xil_In32(config_reg_address) | 0x2));
 }
 
-void eddy_current_sensor_set_sample_rate(uint32_t base_addr, double sample_rate)
+void eddy_current_sensor_trigger_on_pwm_both(uint32_t base_addr)
 {
-    uint8_t divider = (uint8_t)(500000 / sample_rate);
+    // Get the current value of the config register and set both the pwm_high and pwm_low trigger bits
+    uint32_t config_reg_address = base_addr + (3 * sizeof(uint32_t));
+    Xil_Out32(config_reg_address, (Xil_In32(config_reg_address) | 0x3));
+}
 
-    eddy_current_sensor_set_divider(base_addr, divider - 1);
+void eddy_current_sensor_trigger_on_pwm_clear(uint32_t base_addr)
+{
+    // Get the current value of the config register and clear both the pwm_high and pwm_low trigger bits
+    uint32_t config_reg_address = base_addr + (3 * sizeof(uint32_t));
+    Xil_Out32(config_reg_address, (Xil_In32(config_reg_address) & ~0x3));
 }
 
-void eddy_current_sensor_set_divider(uint32_t base_addr, uint8_t divider)
+void eddy_current_sensor_set_timing(uint32_t base_addr, uint32_t sclk_freq_khz, uint32_t propogation_delay_ns)
 {
-    Xil_Out32(base_addr + (2 * sizeof(uint32_t)), divider);
+    // 10 MHz is max frequency (faster frequencies limited by diff/single transceivers)
+    //   should give sclk_cnt = 10 axi cycles
+    if (sclk_freq_khz > 10000) {
+        sclk_freq_khz = 10000;
+    }
+
+    // 500 kHz is min frequency (slower frequencies won't complete in a PWM carrier cycle)
+    //   should give sclk_cnt = 200 axi cycles
+    if (sclk_freq_khz < 500) {
+        sclk_freq_khz = 500;
+    }
```

Comment:
Let's remove the comments which give the raw sclk_cnt values since these are a function of the FPGA clock frequency. If we change the FPGA clock freq, these comments go out of date.